lib/yabeda/datadog/units.rb

```diff
@@ -4,31 +4,44 @@ module Yabeda
-                   "response", "message", "payload", "timeout", "datagram", "route",
-                   "session", "process", "thread", "host", "node", "fault", "service",
-                   "instance", "cpu", "file", "inode", "sector", "block", "buffer",
-                   "error", "read", "write", "occurrence", "event", "time", "unit",
-                   "operation", "item", "task", "worker", "resource", "garbage collection",
-                   "email", "sample", "stage", "monitor", "location", "check", "attempt",
-                   "device", "update", "method", "job", "container", "execution",
-                   "throttle", "invocation", "user", "success", "build", "prediction",
-                   "table", "index", "lock", "transaction", "query", "row", "key",
-                   "command", "offset", "record", "object", "cursor", "assertion", "scan",
-                   "document", "shard", "flush", "merge", "refresh", "fetch", "column",
-                   "commit", "wait", "ticket", "question", "hit", "miss", "eviction",
-                   "get", "set", "dollar", "cent", "page", "split", "hertz", "kilohertz",
-                   "megahertz", "gigahertz", "entry", "degree celsius", "degree fahrenheit",
-                   "nanocore", "microcore", "millicore", "core", "kilocore", "megacore",
-                   "gigacore", "teracore", "petacore", "exacore",].freeze
+      AVAILABLE = [
+        "bit", "byte", "kibibyte", "mebibyte", "gibibyte", "tebibyte",
+        "pebibyte", "exbibyte", "nanosecond", "microsecond", "millisecond",
+        "second", "minute", "hour", "day", "week", "percent_nano", "percent",
+        "apdex", "fraction", "connection", "request", "packet", "segment",
+        "response", "message", "payload", "timeout", "datagram", "route",
+        "session", "process", "thread", "host", "node", "fault", "service",
+        "instance", "cpu", "file", "inode", "sector", "block", "buffer",
+        "error", "read", "write", "occurrence", "event", "time", "unit",
+        "operation", "item", "task", "worker", "resource", "garbage collection",
+        "email", "sample", "stage", "monitor", "location", "check", "attempt",
+        "device", "update", "method", "job", "container", "execution",
+        "throttle", "invocation", "user", "success", "build", "prediction",
+        "table", "index", "lock", "transaction", "query", "row", "key",
+        "command", "offset", "record", "object", "cursor", "assertion", "scan",
+        "document", "shard", "flush", "merge", "refresh", "fetch", "column",
+        "commit", "wait", "ticket", "question", "hit", "miss", "eviction",
+        "get", "set", "dollar", "cent", "page", "split", "hertz", "kilohertz",
+        "megahertz", "gigahertz", "entry", "degree celsius", "degree fahrenheit",
+        "nanocore", "microcore", "millicore", "core", "kilocore", "megacore",
+        "gigacore", "teracore", "petacore", "exacore",
+      ].map { |unit| [unit, unit] }.to_h.freeze
```

Comment:
You reinvented [`Set`](https://ruby-doc.org/stdlib-2.6/libdoc/set/rdoc/Set.html) here. ;-)

Unfortunately, sets doesn't have `[]` method on instances so readability of checks will suffer if replace this `Hash` to `Set`.